### PR TITLE
Check rebind does not effect catalog

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

File: docs/guide/ops/catalog/index.md [hunk 1]
```diff
@@ -302,7 +302,14 @@ The `brooklyn` CLI includes several commands for working with the catalog.
 * `--catalogInitial <file.bom>` will set the catalog items to use on first run,
   on a catalog reset, or if persistence is off
 
-If [persistence](../persistence/) is enabled, catalog additions will remain between runs.
+If `--catalogInitial` is not specified, the default initial catalog at `brooklyn/default.catalog.bom` will be used.
```
Comment (neykov) [inline docs/guide/ops/catalog/index.md]: Worth mentioning (or focusing on) that the default catalog will scan the classpath for catalog items.

File: usage/launcher/src/test/java/brooklyn/launcher/BrooklynLauncherRebindCatalogTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        catalog.addItems(new ResourceUtils(this).getResourceAsString(CATALOG_ADDITIONS));
+
+        assertCatalogConsistsOfIds(catalog.getCatalogItems(), EXPECTED_ADDED_IDS);
+
+        launcher.terminate();
+
+        BrooklynLauncher newLauncher = newLauncherForTests(persistenceDir);
+        newLauncher.start();
+        assertCatalogConsistsOfIds(newLauncher.getServerDetails().getManagementContext().getCatalog().getCatalogItems(), EXPECTED_DEFAULT_IDS);
```
Comment (neykov) [inline usage/launcher/src/test/java/brooklyn/launcher/BrooklynLauncherRebindCatalogTest.java]: If rebinding against previous state, shouldn't we get `EXPECTED_ADDED_IDS` here?

Comment (nakomis) [inline usage/launcher/src/test/java/brooklyn/launcher/BrooklynLauncherRebindCatalogTest.java]: Good catch. Annoyingly, the test was passing as I'd forgotten to set the persistMode

**Comment by nakomis** [comment]
PR comments addressed. Also added apache header to test class, which was causing rat failure

File: usage/launcher/src/test/java/brooklyn/launcher/BrooklynLauncherRebindCatalogTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        Assert.assertEquals(((Collection) catalog.getCatalogItems()).size(), 0);
+
+        catalog.addItems(new ResourceUtils(this).getResourceAsString(CATALOG_ADDITIONS));
+
+        assertCatalogConsistsOfIds(catalog.getCatalogItems(), EXPECTED_ADDED_IDS);
+
+        launcher.terminate();
+
+        BrooklynLauncher newLauncher = newLauncherForTests(persistenceDir);
+        newLauncher.start();
```
Comment (aledsage) [inline usage/launcher/src/test/java/brooklyn/launcher/BrooklynLauncherRebindCatalogTest.java]: what tears down this management context at the end of the test? It looks like it is left running.

**Comment by aledsage** [comment]
Thanks @nakomis - I'll merge this now, and look at the test to see if it needs extra code to terminate the brooklyn launcher.

**Pull request merged**